src/rs_distributions/distributions/rice.py

```diff
@@ -0,0 +1,215 @@
+        batch_shape = self.nu.shape
+        super().__init__(batch_shape, validate_args=validate_args)
+        self._irsample = RiceIRSample().apply
+
+    def _log_bessel_i0(self, x):
+        return torch.log(torch.special.i0e(x)) + torch.abs(x)
+
+    def _log_bessel_i1(self, x):
+        return torch.log(torch.special.i1e(x)) + torch.abs(x)
+
+    def _laguerre_half(self, x):
+        return (1.0 - x) * torch.exp(
+            x / 2.0 + self._log_bessel_i0(-0.5 * x)
+        ) - x * torch.exp(x / 2.0 + self._log_bessel_i1(-0.5 * x))
+
+    def log_prob(self, value):
+        """
+        Compute the log-probability of the given values under the Rice distribution
+
+        ```
+        Rice(x | nu, sigma) = \
+            x * sigma**-2 * exp(-0.5 * (x**2 + nu**2) * sigma ** -2) * I_0(x * nu * sigma **-2)
+        ```
+
+        Args:
+            value (Tensor): The values at which to evaluate the log-probability
+
+        Returns:
+            Tensor: The log-probabilities of the given values
+        """
+        nu, sigma = self.nu, self.sigma
+
+        if self._validate_args:
+            self._validate_sample(value)
+
+        log_sigma = torch.log(sigma)
+        x = value
+        log_x = torch.log(value)
+        log_nu = torch.log(nu)
+        i0_arg = torch.exp(log_x + log_nu - 2.0 * log_sigma)
+
+        log_prob = log_x - 2.0 * log_sigma - 0.5 * (x * x + nu * nu) / (sigma * sigma)
+        log_prob += self._log_bessel_i0(i0_arg)
+
+        return log_prob
+
+    def sample(self, sample_shape=torch.Size()):
+        """
+        Generate random samples from the Rice distribution
+
+        Args:
+            sample_shape (torch.Size, optional): The shape of the samples to generate.
+            Default is an empty shape
+
+        Returns:
+            Tensor: The generated random samples
+        """
+        shape = self._extended_shape(sample_shape)
+        nu, sigma = self.nu, self.sigma
+        nu = nu.expand(shape)
+        sigma = sigma.expand(shape)
+        with torch.no_grad():
+            A = torch.normal(nu, sigma)
+            B = torch.normal(torch.zeros_like(nu), sigma)
+            z = torch.sqrt(A * A + B * B)
+            return z
+
+    @property
+    def mean(self):
+        """
+        Compute the mean of the Rice distribution
+
+        Returns:
+            Tensor: The mean of the distribution.
+        """
+        sigma = self.sigma
+        nu = self.nu
+        mean = (
+            sigma
+            * math.sqrt(math.pi / 2.0)
+            * self._laguerre_half(-0.5 * (nu / sigma) ** 2)
+        )
+        return mean
+
+    @property
+    def variance(self):
+        """
+        Compute the variance of the Rice distribution
+
+        Returns:
+            Tensor: The variance of the distribution
+        """
+        sigma = self.sigma
+        nu = self.nu
+        variance = (
+            2 * sigma**2.0
+            + nu**2.0
+            - 0.5
+            * np.pi
```

Comment:
use `torch.pi`?

Reply:
Resolved in this comment: https://github.com/rs-station/distributions/pull/4/files/44e055764ca277d99a69f1fa207533d0f81c3bf7#r1579539230